cranelift/codegen/src/ir/entities.rs

```diff
@@ -48,6 +48,16 @@ impl Block {
             None
         }
     }
+
+    /// TODO
+    pub fn from_value(v: Value) -> Option<Self> {
+        Self::with_number(v.0)
+    }
+
+    /// TODO
+    pub fn as_value(self) -> Option<Value> {
+        Value::with_number(self.0)
+    }
```

Comment:
This is a gross hack. `Block` and `Value` are two entirely distinct types. They should never be mixed.

Reply:
Maybe you could have a `BlockOrValue` type just for `BlockWithArgs` and then ensure that this type never leaks out of the `BlockWithArgs`?

Reply:
I appreciate how quick you are to share your insights on Cranelift issues and PRs. Your comments are usually helpful and lead to making Cranelift better for everyone. In this case, though, this is not the most helpful comment.

Before commenting on code where I'm not satisfied with its quality, I try to ask myself: Why did the author choose this way of expressing this idea? Maybe they know it's not ideal but they needed a placeholder while they're focusing on other parts of the problem. This case looks like a placeholder, given the empty "TODO" comments. Since this PR is still in "draft" status, placeholders are fine. They're only an issue if we merge something to main that'll cause us problems later.

It's useful to open draft PRs to run CI against work-in-progress. They're also an opportunity to get early feedback, but preferably when the author has specifically requested it. Even then, that feedback should be focused on the high-level ideas, not the implementation details, which are likely to change as the patch matures. (I aspire to follow the "Three-Phase Contribution Review" model described in "[The Gentle Art of Patch Review](https://sage.thesharps.us/2014/09/01/the-gentle-art-of-patch-review/)".)

So, while I agree with you that this PR probably shouldn't be merged in exactly this form, it's too early to critique this implementation detail with the degree of severity of your first reaction. Your `BlockOrValue` suggestion is a good one, but Trevor may have equally good alternatives in mind.

It's good to suggest better alternatives, and even good to note concerns. For example, you could say, "We'll have to carefully audit anywhere that the `Block` and `Value` types are allowed to mix, and in the final version of this PR I'd prefer to avoid that." Just keep in mind that the patch's author may already know that, especially for draft PRs, and that they may even have good reasons for it.

Reply:
I agree I could have reacted much better. I will keep this in mind for the future.